mailet/standard/src/main/java/org/apache/james/transport/mailets/ReplaceContent.java

```diff
@@ -88,263 +94,287 @@
+        String[] patternArray = StringUtils.split(patternProcessing, "/,/");
         
-        PatternList patternList= new PatternList();
+        ImmutableList.Builder<ReplacingPattern> patternList= ImmutableList.builder();
         for (String aPatternArray : patternArray) {
-            PatternBean o = getPattern(aPatternArray);
-            patternList.getPatterns().add(o.getPatterns());
-            patternList.getSubstitutions().add(o.getSubstitutions());
-            patternList.getFlags().add(o.getFlag());
+            patternList.add(getPattern(aPatternArray));
         }
-        
-        return patternList;
+        return patternList.build();
     }
 
-    protected static PatternList getPatternsFromStream(InputStream stream, String charset) throws MailetException, IOException {
-        PatternList patternList= new PatternList();
-        BufferedReader reader = new BufferedReader(charset != null ? new InputStreamReader(stream, charset) : new InputStreamReader(stream));
-        //BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream("q:\\correzioniout"), "utf-8"));
-        
-        String line;
-        while ((line = reader.readLine()) != null) {
+    protected static List<ReplacingPattern> getPatternsFromStream(InputStream stream, String charset) throws MailetException, IOException {
+        ImmutableList.Builder<ReplacingPattern> patternList= ImmutableList.builder();
+        for (String line: IOUtils.readLines(stream, charset)) {
             line = line.trim();
-            if (line.length() > 0 && !line.startsWith("#")) {
-                if (line.length() < 2 && !line.startsWith("/") && !line.endsWith("/")) throw new MailetException("Invalid expression: " + line);
-                PatternBean o = getPattern(line.substring(1, line.length() - 1));
-                patternList.getPatterns().add(o.getPatterns());
-                patternList.getSubstitutions().add(o.getSubstitutions());
-                patternList.getFlags().add(o.getFlag());
+            if (!Strings.isNullOrEmpty(line) && !line.startsWith("#")) {
+                ensurePatternIsValid(line);
+                patternList.add(getPattern(line.substring(1, line.length() - 1)));
             }
         }
-        reader.close();
-        return patternList;
+        return patternList.build();
     }
-    
+
+    private static void ensurePatternIsValid(String pattern) throws MailetException {
+        if (pattern.length() < 2 || (!pattern.startsWith("/") && !pattern.endsWith("/"))) {
+            throw new MailetException("Invalid expression: " + pattern);
+        }
+    }
+
+    private static ReplacingPattern getPattern(String line) throws MailetException {
+        String[] pieces = StringUtils.split(line, "/");
+        if (pieces.length < 3) {
+            throw new MailetException("Invalid expression: " + line);
+        }
+        return new ReplacingPattern(Pattern.compile(pieces[0], extractOptions(pieces[2])), 
```

Comment:
you could name the parts by using constant like that : `static final int PATTERN = 0;`